src/mongocxx/private/libbson.cpp

```diff
@@ -18,34 +18,40 @@ namespace mongocxx {
 MONGOCXX_INLINE_NAMESPACE_BEGIN
 namespace libbson {
 
-static void doc_to_bson_t(const bsoncxx::document::view& doc, bson_t* bson) {
-    bson_init_static(bson, doc.data(), doc.length());
-}
+namespace {
 
-static void optional_doc_to_bson_t(const stdx::optional<bsoncxx::document::view>& doc,
-                                   bson_t* bson) {
-    if (doc) {
-        doc_to_bson_t(*doc, bson);
-    }
+void doc_to_bson_t(const bsoncxx::document::view& doc, bson_t* bson) {
+    bson_init_static(bson, doc.data(), doc.length());
 }
 
-scoped_bson_t::scoped_bson_t(const stdx::optional<bsoncxx::document::view>& doc)
-    : _is_initialized(doc) {
-    optional_doc_to_bson_t(doc, &_bson);
-}
+}  // namespace
 
-scoped_bson_t::scoped_bson_t(const bsoncxx::document::view& doc) : _is_initialized(true) {
+scoped_bson_t::scoped_bson_t(bsoncxx::document::view_or_value doc)
+    : _is_initialized(true), _doc(std::move(doc)) {
     doc_to_bson_t(doc, &_bson);
 }
 
-void scoped_bson_t::init_from_static(const stdx::optional<bsoncxx::document::view>& doc) {
-    _is_initialized = static_cast<bool>(doc);
-    optional_doc_to_bson_t(doc, &_bson);
+scoped_bson_t::scoped_bson_t(bsoncxx::stdx::optional<bsoncxx::document::view_or_value> doc)
+    : _is_initialized(doc) {
+    if (doc) {
+        doc_to_bson_t(*doc, &_bson);
+        _doc = doc;
```

Comment:
Do we want to swap the order of these lines, then call `doc_to_bson_t(*_doc, &_bson);` instead? Do the views/values/v_or_v's passed to `scoped_bson_t` have to outlive the `scoped_bson_t`?